packages/core/src/WorkflowExecute.ts

```diff
@@ -360,7 +360,7 @@ export class WorkflowExecute {
 		startNodes = handleCycles(graph, startNodes, trigger);
 
 		// 6. Clean Run Data
-		const newRunData: IRunData = cleanRunData(runData, graph, startNodes);
+		runData = cleanRunData(runData, graph, startNodes);
```

Comment:
`runData` is already a let binding. No need to create another const. This also fixes the issue that on line 367 I was passing the old run data into `recreateNodeExecutionStack` instead of the new run data.